rir/src/compiler/translations/rir_2_pir/rir_2_pir_compiler.cpp

```diff
@@ -124,29 +126,79 @@ void Rir2PirCompiler::compileClosure(Closure* closure,
     return fail();
 }
 
+bool MEASURE_COMPILER_PERF = getenv("PIR_MEASURE_COMPILER") ? true : false;
+class CompilerPerf {
+    std::unordered_map<std::string, double> passTimer;
+
+  public:
+    void addTime(const std::string& name, double time) {
+        if (!passTimer.count(name))
+            passTimer[name] = 0;
+        passTimer.at(name) += time;
+    }
+
+    ~CompilerPerf() {
+        if (!MEASURE_COMPILER_PERF)
+            return;
+        std::map<double, std::string> ordered;
+        double total = 0;
+        for (auto t : passTimer) {
+            while (ordered.count(t.second))
+                t.second += 1e-20;
+            ordered.emplace(t.second, t.first);
+            total += t.second;
+        }
+
+        std::cerr << "=== COMPILER perf Breakdown:\n";
+        for (auto t : ordered)
+            std::cerr << "" << std::setw(24) << t.second << "\t" << t.first
+                      << "\n";
+        std::cerr << "" << std::setw(24) << "total"
+                  << "\t" << total << "\n";
+    }
+};
+std::chrono::time_point<std::chrono::high_resolution_clock> startTime;
+std::chrono::time_point<std::chrono::high_resolution_clock> endTime;
+CompilerPerf PERF;
+
 void Rir2PirCompiler::optimizeModule() {
     logger.flush();
     size_t passnr = 0;
     for (auto& translation : translations) {
         module->eachPirClosure([&](Closure* c) {
             c->eachVersion([&](ClosureVersion* v) {
+                if (MEASURE_COMPILER_PERF)
```

Comment:
True that we can use it with a raw pir_debug, but I do not see why we would want to profile the logger, so, I would measure the time without including the log times